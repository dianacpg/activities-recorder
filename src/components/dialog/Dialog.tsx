// Style
import styles from "./styles/dialog.module.scss";
// Components
import Button from "../button";

export interface DialogProps {
  title: string;
  description?: string;
  onConfirm: () => void;
  onCancel?: () => void;
}

const Dialog = ({ title, description, onConfirm, onCancel }: DialogProps) => (
  <div data-testid="dialog" className={styles["dialog"]}>
    <div className={styles["dialog__content"]}>
      <h2>{title}</h2>
      <p>{description}</p>
      <footer className={styles["dialog__footer"]} data-testid="custom-element">
        <Button data-testid="cancel-dialog" onClick={onCancel}>
          Cancel
        </Button>
        <Button data-testid="confirm-dialog" onClick={onConfirm}>
          Confirm
        </Button>
      </footer>
    </div>
  </div>
);

export default Dialog;
