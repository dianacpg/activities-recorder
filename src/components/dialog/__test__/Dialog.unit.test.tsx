// Test
import { describe, it, expect, vi } from "vitest";
import { render, fireEvent } from "@testing-library/react";
// Component
import Dialog, { DialogProps } from "../Dialog";

describe("Dialog component", () => {
  const defaultProps: DialogProps = {
    title: "Test Dialog",
    description: "This is a test dialog.",
    onConfirm: vi.fn(),
    onCancel: vi.fn(),
  };

  it("renders Dialog component", () => {
    expect(render(<Dialog {...defaultProps} />)).toMatchSnapshot();
  });

  it("calls onConfirm when Confirm button is clicked", () => {
    const { getByTestId } = render(<Dialog {...defaultProps} />);
    fireEvent.click(getByTestId("confirm-dialog"));
    expect(defaultProps.onConfirm).toHaveBeenCalled();
  });

  it("calls onCancel when Cancel button is clicked", () => {
    const { getByTestId } = render(<Dialog {...defaultProps} />);
    fireEvent.click(getByTestId("cancel-dialog"));
    expect(defaultProps.onCancel).toHaveBeenCalled();
  });
});
