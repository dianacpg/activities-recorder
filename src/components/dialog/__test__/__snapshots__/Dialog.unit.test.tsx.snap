// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog component > renders Dialog component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dialog"
        data-testid="dialog"
      >
        <div
          class="dialog__content"
        >
          <h2>
            Test Dialog
          </h2>
          <p>
            This is a test dialog.
          </p>
          <footer
            class="dialog__footer"
          >
            <button
              class="button"
              data-testid="cancel-dialog"
            >
              Cancel
            </button>
            <button
              class="button"
              data-testid="confirm-dialog"
            >
              Confirm
            </button>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dialog"
      data-testid="dialog"
    >
      <div
        class="dialog__content"
      >
        <h2>
          Test Dialog
        </h2>
        <p>
          This is a test dialog.
        </p>
        <footer
          class="dialog__footer"
        >
          <button
            class="button"
            data-testid="cancel-dialog"
          >
            Cancel
          </button>
          <button
            class="button"
            data-testid="confirm-dialog"
          >
            Confirm
          </button>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
